### PR TITLE
missing `#![no_std]` in core leading to std

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**

### CODE REVIEW

**Comment by hussein-aitlahcen** [comment]
fixes #374

**Comment by Robbepop** [comment]
Oh wow!
I actually had to check the codebase locally to make sure that this is real.
Thanks for the fix!

File: core/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![no_std]
```
Comment (Robbepop) [inline core/src/lib.rs:1]:
```suggestion
#![cfg_attr(not(feature = "std"), no_std)]
```

Comment (Robbepop) [reply core/src/lib.rs:1]: We don't want to enable `no_std` when the `std` feature is enabled as with all crates.

Comment (hussein-aitlahcen) [reply core/src/lib.rs]: indeed

**Pull request merged**